### PR TITLE
fix #271445: allow MusicXML export with .xml extension

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
See also https://musescore.org/en/node/14391#comment-876568 ff.

### CODE REVIEW

**Comment by Jojo-Schmitz** [comment]
I still don't.
Although hiding it behind an advanced preference might not be the most brilliant idea, advanced user tend to know that they want or need and how to achieve it, it is the non-advanced ones that are more likely stumble accross the need and struggle with the manual renaming.

**Comment by Jojo-Schmitz** [comment]
See for exaple https://musescore.org/en/node/27451 what proples uses have with a simple chage of file extension, this does come very frequently. So frequently that allowing to open `*.mscz,` files should also be made possible somehow, of course with proper warnings and all that...

File: mscore/file.cpp [hunk 1]
```diff
@@ -1949,7 +1953,7 @@ bool MuseScore::saveAs(Score* cs_, bool saveCopy, const QString& path, const QSt
                   writeSessionFile(false);
                   }
             }
-      else if (ext == "musicxml") {
+      else if ((ext == "musicxml") || (ext == "xml")) {
```
Comment (Jojo-Schmitz) [inline mscore/file.cpp]: this line alone will allow uses to manually change musicxml to xml and make the .-o commandline option work

File: mscore/preferences.cpp [hunk 2]
```diff
@@ -98,6 +98,7 @@ void Preferences::init(bool storeInMemoryOnly)
             {PREF_EXPORT_MP3_BITRATE,                              new IntPreference(128, false)},
             {PREF_EXPORT_MUSICXML_EXPORTBREAKS,                    new EnumPreference(QVariant::fromValue(MusicxmlExportBreaks::ALL), false)},
             {PREF_EXPORT_MUSICXML_EXPORTLAYOUT,                    new BoolPreference(true, false)},
+            {PREF_EXPORT_MUSICXML_OUTDATED,                        new BoolPreference(false)},
```
Comment (Jojo-Schmitz) [inline mscore/preferences.cpp]: not sure whether `false` is the right default, it is the less advanced (Windows) users that have a problem with the renaming.

**Comment by jeetee** [comment]
My vote is (still) for not making it an option, but available as "legacy" support. But I'll take the advanced setting if that's what's needed to get it in.

Cfr. programs like word that still allow you to "Save As.../Export" to the Word97 format and then throw a message box in your face warning you that you will not be future proof and lose formatting etc…

While I do agree on attempting to be progressive in embracing the new standard, you can't ignore the fact that many of our first time users come from different software packages and (especially at first) will move a score back and forth between those versions.
Only a part of those users will complain onto the forum that we could import musicxml.xml from their software, but export "incompatible" musicxml.musicxml (if they're aware of the extention at all!).

My stance would be more towards the new one if maintaining the legacy file extension would actually lead to additional compatibility code.

File: mscore/file.cpp [hunk 3]
```diff
@@ -1679,6 +1679,8 @@ void MuseScore::exportFile()
       fl.append(tr("Standard MIDI File") + " (*.mid)");
       fl.append(tr("Compressed MusicXML File") + " (*.mxl)");
       fl.append(tr("Uncompressed MusicXML File") + " (*.musicxml)");
+      if (preferences.getBool(PREF_EXPORT_MUSICXML_OUTDATED))
+            fl.append(tr("Uncompressed MusicXML File (oudated)") + " (*.xml)");
```
Comment (lasconic) [inline mscore/file.cpp]: typo

Comment (lasconic) [reply mscore/file.cpp]: oudated -> outdated

**Comment by MarcSabatella** [comment]
That is my vote as well.  No option to clutter the inerface and never be discovered by the people who need it just show both formats in the list with the old one marked "legacy" or some such.

File: mscore/file.cpp (hunk 3 above)
Comment (Jojo-Schmitz) [reply mscore/file.cpp]: fixed

**Comment by Jojo-Schmitz** [comment]
OK, options removed. Reduces the change to 1 changed and 2 added lines

**Comment by Jojo-Schmitz** [comment]
This is being requested and discussed again in https://musescore.org/en/node/271445#comment-954085 ff.

File: mscore/preferences.cpp (hunk 2 above)
Comment (Jojo-Schmitz) [reply mscore/preferences.cpp]: That'd be possible, but would be quite confusing and likely to create traffic in the forum. So far there's only one setting where the default is platform dependant, `nativeDialogs`.

File: mscore/file.cpp (hunk 1 above)
Comment (dmitrio95) [reply mscore/file.cpp]: Let's leave this line alone then here? Adding another preference for this probably rare thing might be not appropriate indeed.

Comment (Jojo-Schmitz) [reply mscore/file.cpp]: This line plus the other one from this commit? Or really require uses to manually change musicxml to xml? Esp. windows users, who are having difficulties with this already, not even seeing the extensions by default?

Comment (Jojo-Schmitz) [reply mscore/file.cpp]: it isn't a rare case though, has been requested quite often. Or had been explained to the users that they need to rename manually.

Comment (dmitrio95) [reply mscore/file.cpp]: Well, let's leave this commit alone then, without the preferences. Hopefully that `(outdated)` marking would be enough not to encourage its usage too much

Comment (Jojo-Schmitz) [reply mscore/file.cpp]: So you'll merge that 1st commit? Need me to resubmit this PR without the 2nd commit for that or can you just cherry-pick the 1st one?

Comment (dmitrio95) [reply mscore/file.cpp]: Yes, I can cherry-pick the commit, will do it now.

Comment (Jojo-Schmitz) [reply mscore/file.cpp]: Thanks

**Comment by dmitrio95** [comment]
Cherry-picked the first commit (without hiding this option behind the preference) in https://github.com/musescore/MuseScore/commit/979615972fe4ab5abfe04848738d5890a59a1851.

**Pull request closed without merging**